hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2564,6 +2564,9 @@ public void createS3Bucket(String userName, String s3BucketName)
       }
       metrics.incNumBucketCreates();
       try {
+        metadataManager.getLock().acquireS3BucketLock(s3BucketName);
+        metadataManager.getLock().acquireVolumeLock(
```

Comment:
If getting the volume lock fails, the finally block will still attempt to release it. Same with the S3 bucket lock.

You will either need to use nested try-catch or a boolean flag per lock indicating it was successfully acquired.

Reply:
Many of these bugs around proper releasing on failure have always been in the codebase. However this is a good time to fix them.

Reply:
I see the only case for failing is with RunTimeException. So, do we still need the flags?

Reply:
On a side note: once we use new HA code this will be cleaned up, so not considered much refactoring here.